### ISSUE TITLE
systemd-nspawn --private-users fails if a second /proc is not present.

### ISSUE THREAD

**Issue by jwwalton** [systemd/systemd#6018]
v233 on 4.10.12. pretty sure but not certain this happened with v232 as well.

```
# systemd-nspawn --register=false --private-users=yes -D test /usr/bin/id 
Spawning container test on /tmp/foobar/test.
Press ^] three times within 1s to kill container.
Child died too early.
Selected user namespace base 1461518336 and range 65536.
Failed to mount proc on /proc (MS_NOSUID|MS_NODEV|MS_NOEXEC ""): Operation not permitted
```

```
# mount -t proc proc /mnt
                                                                                                                                                                                                                                                                  
# systemd-nspawn --register=false --private-users=yes -D test /usr/bin/id
Spawning container test on /tmp/foobar/test.
Press ^] three times within 1s to kill container.
Selected user namespace base 1461518336 and range 65536.
uid=0(root) gid=0(root) groups=0(root)
Container test exited successfully.
```

```
# umount /mnt

# systemd-nspawn --register=false --private-users=yes -D test /usr/bin/id
Spawning container test on /tmp/foobar/test.
Press ^] three times within 1s to kill container.
Child died too early.
Selected user namespace base 1461518336 and range 65536.
Failed to mount proc on /proc (MS_NOSUID|MS_NODEV|MS_NOEXEC ""): Operation not permitted
```

**Comment by poettering** [comment]
hmm, strange. Have you played around with the propagation settings of your local host mounts? Can you provide the contents of /proc/self/mountinfo right before the failed and the successful run?

**Comment by teknoraver** [comment]
Probably I have a similar issue, I hope it can help:
```
root@turbo:/var/lib/machines/fedora# systemd-nspawn --network-macvlan=enp9s0 -U --machine=fedora
Spawning container fedora on /var/lib/machines/fedora.
Press ^] three times within 1s to kill container.
Selected user namespace base 702021632 and range 65536.
Failed to mount n/a on /var/lib/machines/fedora/sys/fs/selinux (MS_BIND ""): No such file or directory
Failed to mount n/a on /var/lib/machines/fedora/sys/fs/selinux (MS_RDONLY|MS_NOSUID|MS_NODEV|MS_NOEXEC|MS_REMOUNT|MS_BIND ""): Invalid argument
[root@fedora ~]# cat /proc/self/mountinfo
225 202 259:2 /var/lib/machines/fedora / rw,noatime master:1 - ext4 /dev/root rw,errors=remount-ro,data=ordered
226 225 0:30 / /sys ro,nosuid,nodev,noexec,relatime - tmpfs tmpfs rw,mode=755,uid=702021632,gid=702021632
227 225 0:40 / /dev rw,nosuid - tmpfs tmpfs rw,mode=755,uid=702021632,gid=702021632
228 227 0:41 / /dev/shm rw,nosuid,nodev - tmpfs tmpfs rw,uid=702021632,gid=702021632
229 227 0:43 / /dev/pts rw,nosuid,noexec,relatime - devpts devpts rw,gid=702021637,mode=620,ptmxmode=666
230 227 0:14 /0 /dev/console rw,nosuid,noexec,relatime master:4 - devpts devpts rw,gid=5,mode=620,ptmxmode=000
231 225 0:42 / /run rw,nosuid,nodev - tmpfs tmpfs rw,mode=755,uid=702021632,gid=702021632
232 231 0:15 /systemd/nspawn/propagate/fedora /run/systemd/nspawn/incoming ro master:11 - tmpfs tmpfs rw,mode=755
163 225 0:44 / /proc rw,nosuid,nodev,noexec,relatime - proc proc rw
167 163 0:44 /sys /proc/sys ro,nosuid,nodev,noexec,relatime - proc proc rw
168 167 0:44 /sys/net /proc/sys/net rw,nosuid,nodev,noexec,relatime - proc proc rw
169 163 0:44 /sysrq-trigger /proc/sysrq-trigger ro,nosuid,nodev,noexec,relatime - proc proc rw
171 225 0:45 / /tmp rw - tmpfs tmpfs rw,uid=702021632,gid=702021632
173 226 0:46 /block /sys/block ro,nosuid,nodev,noexec,relatime - sysfs sysfs ro
174 226 0:46 /bus /sys/bus ro,nosuid,nodev,noexec,relatime - sysfs sysfs ro
175 226 0:46 /class /sys/class ro,nosuid,nodev,noexec,relatime - sysfs sysfs ro
176 226 0:46 /dev /sys/dev ro,nosuid,nodev,noexec,relatime - sysfs sysfs ro
177 226 0:46 /devices /sys/devices ro,nosuid,nodev,noexec,relatime - sysfs sysfs ro
178 226 0:46 /kernel /sys/kernel ro,nosuid,nodev,noexec,relatime - sysfs sysfs ro
172 226 0:57 / /sys/fs/cgroup rw,nosuid,nodev,noexec - tmpfs tmpfs rw,mode=755,uid=702021632,gid=702021632
179 172 0:22 / /sys/fs/cgroup/pids rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,pids
180 172 0:24 / /sys/fs/cgroup/cpu rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,cpu
181 172 0:26 / /sys/fs/cgroup/memory rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,memory
182 172 0:21 / /sys/fs/cgroup/devices rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,devices
183 172 0:25 / /sys/fs/cgroup/rdma rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,rdma
184 172 0:23 / /sys/fs/cgroup/blkio rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,blkio
185 172 0:18 / /sys/fs/cgroup/systemd rw,nosuid,nodev,noexec,relatime - cgroup cgroup rw,xattr,release_agent=/lib/systemd/systemd-cgroups-agent,name=systemd
189 167 0:42 /proc-sys-kernel-random-boot-id//deleted /proc/sys/kernel/random/boot_id ro,nosuid,nodev - tmpfs tmpfs rw,mode=755,uid=702021632,gid=702021632
190 163 0:42 /kmsg//deleted /proc/kmsg rw,nosuid,nodev - tmpfs tmpfs rw,mode=755,uid=702021632,gid=702021632
```

**Comment by poettering** [comment]
I guess this has been fixed by b71a0192c040f585397cfc6fc2ca025bf839733d
